apps/src/applab/designElements/library.js

```
@@ -160,6 +160,12 @@ export default {
     }
     // Unknown elements are expected. Return null because we don't know type.
     if (allowUnknown) {
+      console.warn(
+        'Project contains an element with an unknown type' +
+          `\nType: ${element.tagName}` +
+          `\nId: ${element.id}` +
+          `\nClass: ${element.className}`
+      );
       return null;
     }
     throw new Error('unknown element type');
```

Do we ever want to throw on unknown elements? If not, maybe we don't need the `allowUnknown` parameter and we can always just console.warn and return null?

---

I'll look a bit deeper. It would be nice to remove this throw.

---

Took a deeper look - there are a lot of areas that rely on this returning something. Additionally, there are a lot of areas that throw when there is a bad design element. I've created a task to look into a better way to approach our strategy on handling malformed design elements.
https://codedotorg.atlassian.net/browse/STAR-556